### PR TITLE
Scrapping Script for Housing.com for rental flats

### PR DESCRIPTION
**Pull request by deepalib-cuelogic**
Scrapping Script for Housing.com for rental flats using python requests and lxml.

### CODE REVIEW

**Comment by aloka-cuelogic** [comment]
@sameetb-cuelogic 

please review and merge

File: housing/housing.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+from lxml import html
+import requests
+import time
+
+
+HEADERS = {'User-Agent':
+           'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:38.0) Gecko/20100101 Firefox/38.0'}
+
+
+def scrap_housing():
```
Comment (sameetb-cuelogic) [inline housing/housing.py]: rename to 'parse_content' or 'parse'

File: housing/housing.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    place = raw_input('Enter city: ').lower()
```
Comment (sameetb-cuelogic) [inline housing/housing.py]: if this script is to be called by other scripts, like the crawler, then the arguments need to be command line with this function accepting the params as arguments

use argparse in the **main** to parse the args

a script importing this one can directly us the function then

File: housing/housing.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    area = raw_input('Enter area (For multiple area enter ',' seprated values): ').lower()
+    AREAS = area.split(',')
```
Comment (sameetb-cuelogic) [inline housing/housing.py]: non-global so needs to be all small caps

File: housing/housing.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    for area in AREAS:
+        page = requests.get(
+            'https://housing.com/in/rent/%s/%s'% (place,area),
+            headers=HEADERS)
+
+        tree = html.fromstring(page.text)
+
+        product_url_list = tree.xpath('//div[@class="heading-content information"]/a[@href]/@href')
+        print product_url_list
```
Comment (sameetb-cuelogic) [inline housing/housing.py]: if printing for logging purposes then mention that in a comment above this line as a reminder to remove later

File: housing/housing.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        for link in product_url_list:
+            time.sleep(10)
```
Comment (sameetb-cuelogic) [inline housing/housing.py:29]: add a comment mentioning reason for sleep

File: housing/housing.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            page = requests.get(link,headers=HEADERS)
+
+            tree = html.fromstring(page.text)
+
+            product_url = tree.xpath('string(//meta[@property="og:url"]/@content)').strip()
+            from_site = tree.xpath('string(//meta[@property="og:site_name"]/@content)').strip()
+            description = tree.xpath('string(//meta[@name="description"]/@content)').strip()
+            price = tree.xpath('string(//div[@class="banner"]/div[1]/div[2]/div[1]/span[2]/text())').strip()
+            property_id = tree.xpath('string(//div[@class="apartment-header"]/div[2]/ul/li[4]/text())').strip()
+            print '\n\n'
```
Comment (sameetb-cuelogic) [inline housing/housing.py:40]: again add a reminder if needed to remove the print statements

**Comment by deepalib-cuelogic** [comment]
@sameetb-cuelogic 

Have implemented commented changes.Please review and merge

**Pull request merged**